### PR TITLE
Add test for DST balancing

### PR DESCRIPTION
**Pull request by jessealama**
References:

+ https://github.com/tc39/proposal-temporal/issues/1791

### CODE REVIEW

**Comment by rwaldron** [comment]
Thanks! When you're ready for review, please ping any of the following folks: @Ms2ger @romulocintra @sarahghp @ryzokuken

**Comment by jessealama** [comment]
cc @ptomato @Ms2ger

**Review by Ms2ger — APPROVED** [review]

File: test/built-ins/Temporal/Duration/compare/twenty-five-hour-day.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+---*/
+
+const tz = TemporalHelpers.springForwardFallBackTimeZone();
+
+// 2000-10-29 is a 25-hour day according to this time zone...
+
+const relativeTo = new Temporal.ZonedDateTime(941187600_000_000_000n, tz);
+
+// confirm that we have rewound one year and one day:
+assert.sameValue('1999-10-29T01:00:00-08:00[Custom/Spring_Fall]', relativeTo.toString());
```
[the remark this answers is not in the document]
Comment (jessealama) [reply test/built-ins/Temporal/Duration/compare/twenty-five-hour-day.js:18]: fixed

File: test/built-ins/Temporal/Duration/compare/twenty-five-hour-day.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.duration.compare
+description: Unbalancing handles DST days with more than hours
```
Comment (ptomato) [inline test/built-ins/Temporal/Duration/compare/twenty-five-hour-day.js:6]:
```suggestion
description: Unbalancing handles DST days with more than 24 hours
```

**Review by ptomato — APPROVED** [review]

**Comment by ptomato** [comment]
This LGTM, but should wait until https://github.com/tc39/proposal-temporal/pull/2026 achieves consensus at TC39.

**Review by ptomato — APPROVED** [review]
This normative change achieved consensus at TC39 today, so I'll merge this now.

**Pull request merged**